src/context/ModalProvider/ModalContainer.tsx

```diff
@@ -1,56 +1,225 @@
+      default: SellAssetSearchModal,
+    })),
+  ),
+)
+
+const BuyTradeAssetSearchModal = makeSuspenseful(
+  lazy(() =>
+    import('components/Modals/TradeAssetSearch/TradeAssetSearchModal').then(
+      ({ BuyTradeAssetSearchModal }) => ({
+        default: BuyTradeAssetSearchModal,
+      }),
+    ),
+  ),
+)
+
+const SellTradeAssetSearchModal = makeSuspenseful(
+  lazy(() =>
+    import('components/Modals/TradeAssetSearch/TradeAssetSearchModal').then(
+      ({ SellTradeAssetSearchModal }) => ({
+        default: SellTradeAssetSearchModal,
+      }),
+    ),
+  ),
+)
+
+const NativeOnboarding = makeSuspenseful(
+  lazy(() =>
+    import('components/Modals/NativeOnboarding/NativeOnboarding').then(({ NativeOnboarding }) => ({
+      default: NativeOnboarding,
+    })),
+  ),
+)
+
+const NftModal = makeSuspenseful(
+  lazy(() =>
+    import('components/Modals/Nfts/NftModal').then(({ NftModal }) => ({
+      default: NftModal,
+    })),
+  ),
+)
+
+const FeedbackAndSupport = makeSuspenseful(
+  lazy(() =>
+    import('components/Modals/FeedbackSupport/FeedbackSupport').then(({ FeedbackAndSupport }) => ({
+      default: FeedbackAndSupport,
+    })),
+  ),
+)
+
+const Snaps = makeSuspenseful(
+  lazy(() =>
+    import('components/Modals/Snaps/Snaps').then(({ Snaps }) => ({
+      default: Snaps,
+    })),
+  ),
+)
+
+const ManageAccountsModal = makeSuspenseful(
+  lazy(() =>
+    import('components/Modals/ManageAccounts/ManageAccountsModal').then(
+      ({ ManageAccountsModal }) => ({
+        default: ManageAccountsModal,
+      }),
+    ),
+  ),
+)
+
+const LedgerOpenAppModal = makeSuspenseful(
+  lazy(() =>
+    import('components/Modals/LedgerOpenApp/LedgerOpenAppModal').then(({ LedgerOpenAppModal }) => ({
+      default: LedgerOpenAppModal,
+    })),
+  ),
+)
+
+export const MODALS: Modals = {
+  receive: ReceiveModal,
+  qrCode: QrCodeModal,
+  send: SendModal,
+  popup: PopupWindowModal,
+  fiatRamps: FiatRampsModal,
+  settings: SettingsModal,
+  keepKeyWipe: WipeModal,
+  backupNativePassphrase: BackupPassphraseModal,
+  mobileWelcomeModal: MobileWelcomeModal,
+  addAccount: AddAccountModal,
+  assetSearch: AssetSearchModal,
+  buyAssetSearch: BuyAssetSearchModal,
+  sellAssetSearch: SellAssetSearchModal,
+  buyTradeAssetSearch: BuyTradeAssetSearchModal,
+  sellTradeAssetSearch: SellTradeAssetSearchModal,
+  nativeOnboard: NativeOnboarding,
+  nft: NftModal,
+  feedbackSupport: FeedbackAndSupport,
+  snaps: Snaps,
+  // Important: Order matters here -This modal must be mounted before the ManageAccountsModal so it can be opened
+  ledgerOpenApp: LedgerOpenAppModal,
+  manageAccounts: ManageAccountsModal,
+} as const
+
+export const modalReducer = (state: ModalState, action: ModalActions<keyof Modals>): ModalState => {
   switch (action.type) {
     case OPEN_MODAL:
-      return { isOpen: true, props: action.props }
+      return {
+        ...state,
+        [action.key]: { isOpen: true, props: action.props },
```

Comment:
The guts of this PR - modals are now indexed per modal in a *global* reducer vs. many reducers (which were useless, as only one reducer was effective)